modules/post/multi/manage/zip.rb

```diff
@@ -0,0 +1,92 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class MetasploitModule < Msf::Post
+
+  include Msf::Post::File
+
+  def initialize(info={})
+    super(update_info(info,
+        'Name'          => 'Multi Manage File Compressor',
+        'Description'   => %q{
+          This module zips a file or a directory. On Linux, it uses the zip command.
+          On Windows, it will try to use remote target's 7Zip if found. If not, it falls
+          back to its own VBScript.
+        },
+        'License'       => MSF_LICENSE,
+        'Author'        => [ 'sinn3r' ],
+        'Platform'      => [ 'win', 'linux' ],
+        'SessionTypes'  => [ 'meterpreter', 'shell' ]
+    ))
+
+    register_options(
+      [
+        OptString.new('DESTINATION', [false, 'The destination path']),
+        OptString.new('SOURCE', [true, 'The directory or file to compress'])
+      ], self.class)
+  end
+
+  def get_program_file_path
+    get_env('ProgramFiles')
```

Comment:
I'll lead this with the statement I do not know how to make it better, but it bothers me that this will fail when it runs in a 32-bit meterpreter on a 64-bit Windows OS containing a 64-bit 7zip because reflection is going to tell meterpreter to look in C:\Program Files (x86)\  rather than C:\Program Files where the 7-zip binary would be (assuming a 64-bit release).  Have we come up with a way around it?  I played a little, and I came up with nothing.  Mostly just curious.